### ISSUE TITLE
Support for custom citation and bibliography styles

### ISSUE THREAD

**Issue by simonbyrne** [ali-ramadhan/DocumenterCitations.jl#22]
Would be useful to be able to customize how the citation and bibliographies are printed.

**Comment by ali-ramadhan** [comment]
I guess we can pick a few styles (ones we want to use) from

https://www.overleaf.com/learn/latex/Bibtex_bibliography_styles
https://www.overleaf.com/learn/latex/Natbib_bibliography_styles
https://www.overleaf.com/learn/latex/Biblatex_bibliography_styles

and if someone wants a specific (official?) style they can open a PR with it? Or we can add a section in the docs showing how to define custom styles via dispatch?

**Comment by kellertuer** [comment]
Maybe it would be easiest to support CSL for styles?

https://github.com/citation-style-language/styles

**Comment by Azzaare** [comment]
Import/export from/to CSL in Bibliography.jl is in the todo list of this summer. I will be back to you once it is done!

**Comment by goerz** [comment]
The newly released [DocumenterCitations.jl v1.0](https://github.com/JuliaDocs/DocumenterCitations.jl/releases/tag/v1.0.0) provides a larger range of [citation styles](https://juliadocs.org/DocumenterCitations.jl/dev/gallery/), including a new default numeric style.

The updated version also allows defining completely custom styles, [as described in the documentation](https://juliadocs.org/DocumenterCitations.jl/dev/internals/#customization); although the methods for this are considered internal, and thus not part of the stable API.

Support for general CSL styles seems like it could be an attractive feature and is being tracked in https://github.com/JuliaDocs/DocumenterCitations.jl/issues/13

**Comment by kellertuer** [comment]
Thanks for the follow up :) I am eager to try this out, since I have been through a few different ways of citations by now in my docs and would love to find one that fits even better than the current ones.

_edit:_ It even covers my favourite way of having a references section per documentation page (yay!) – I will definetly try this out!
